### ISSUE TITLE
Breakpoints don't work in IntelliJ IDEA with Dartium 50 

### ISSUE THREAD

**Issue by zoechi** [dart-lang/sdk#29640]
Execution doesn't halt on breakpoints in web applications (works fine on console apps)

```
Dart VM version: 1.24.0-edge.ca3d90c4cd7155709a7836d0830b1f712925a8cd (Thu May 11 19:41:00 2017) on "linux_x64"
```

Tested with 2017.2 EAP and 2017.1.3 EAP

Related to #28280

**Comment by devoncarew** [comment]
@zoechi, is this just when used from IntelliJ, or does it show up in devtools as well?

**Comment by zoechi** [comment]
@davon I haven't tried with devtools. 
I also since moved away from Linux to Mac and away from bleeding-edge to dev.

**Comment by alexander-doroshko** [comment]
/cc @terrylucas 
If it helps, attached is a [minimal project](https://github.com/dart-lang/sdk/files/1028689/anagram.zip) to reproduce. The project also contains logs of IDE<->Dartium communication both for Dartium 1.23.0 (successful debugging) and 1.24.0-dev.6.0 (no success). The biggest difference I see is that Dartium 1.23.0 sends `"Debugger.scriptParsed"` events, but Dartium 1.24.0-dev.6.0 sends `"Debugger.scriptFailedToParse"` instead.

**Comment by alexander-doroshko** [comment]
Hint: to get a nice files comparison select 2 files in the Project View and right-click -> Compare Files :).
![image](https://cloud.githubusercontent.com/assets/4499789/26446599/620708b2-414d-11e7-9915-960eab7368de.png)

**Comment by alexander-doroshko** [comment]
This issue looks like a 1.24 blocker. I'm adding area-dartium tag. 
If you think it is an IDE-level problem please let me know. But so far it looks more like a Dartium problem caused by Dartium sending `"Debugger.scriptFailedToParse"` events instead of `"Debugger.scriptParsed"`.
Not sure who's the right person to ask. @jacob314? @terrylucas? I need to make sure no actions at the IDE end required.
Thanks for taking a look!

**Comment by devoncarew** [comment]
@terrylucas, I'm tentatively assigning this issue to you - please re-assign if appropriate.

**Comment by terrylucas** [comment]
alexander-doroshko I have a build of dartium with output to stdout how can I capture/see that information?  Webstorm has started my version of dartium (short of me logging to a file).

**Comment by alexander-doroshko** [comment]
Help -> Find Action, type `Registry`, search for `js.debugger.wip.log` property, set it to some log file path. This file will be rewritten on each debug session.

**Comment by terrylucas** [comment]
Found the problem in DartInspectorDebuggerAgent::didParseSource the parameters to this function changed in a flipped way.

Dartium 45:  (const String& scriptId, const Script& parsedScript, CompileResult compileResult)
Dartium 50:  (const String& scriptId, const Script& parsedScript, bool success)

The problem is that CompileResult was an enum 

enum CompileResult { CompileSuccess, CompileError };

Unfortunately, 0 == CompileSuccess and 1 == CompileError    Ugh!

**Comment by whesse** [comment]
Fixed by CL https://codereview.chromium.org/2915643002, rolled in DEPS by https://github.com/dart-lang/sdk/commit/b987a126c541590cd201d54bc1002dd603cc86af

Fix is landed in both 1.24 and 1.25 branches.
